### PR TITLE
Improved Detail View

### LINKED ISSUE #45: Style and add content to shared detail view
Follow up from https://github.com/mozilla/network-pulse/issues/39

When someone ~~clicks on the~~ _follows a_ shared URL the card/thumbnail should show in lightbox.

### PR DESCRIPTION
**Pull request by mmmavis**
This needs to wait for PR https://github.com/mozilla/network-pulse/pull/47 is landed first. When PR https://github.com/mozilla/network-pulse/pull/47 is landed I'll rebase & update this PR.

Fixes #45 - improved detail view

### CODE REVIEW

File: js/pulse.js [hunk 1]
```diff
@@ -204,27 +204,27 @@ var starred = {
 
 /* Detailer View Handler */
```
Comment (mmmavis) [inline js/pulse.js]: changes to this file:

- Renamed `enableDetailView()` to `DetailViewManager` and make it an `object` instead
- Added `init()` method to it
- Pretty much just cloning a project's thumbnail HTML and put it onto the `detail-view` wrapper.
- Made sure `shareBtnClickHandler` is attached to all `share-btn` (thumbnail ones and the one on detail view)

**Comment by mmmavis** [comment]
@acabunoc / @cadecairos plz review PR https://github.com/mozilla/network-pulse/pull/47 first as this is PR is built on top of that. 😉

**Comment by mmmavis** [comment]
Note that the "star" feature doesn't work in detail view. I'm planning to fix it as part of https://github.com/mozilla/network-pulse/issues/51.

**Comment by cadecairos** [comment]
Looks good, works as intended. If I could make one comment, it would be about how I took ten seconds to find the close button for the lightbox (top right of screen). I personally expected it to be near the card.

I might also suggest binding a close action to the escape key or even a click event outside the card view could close it.

That all said, just my opinion, I'd like to hear @kristinashu's thoughts.

screenshot from 2016-07-28 16-07-04

**Comment by kristinashu** [comment]
Good point @cadecairos. It would be nice if the X was closer, or if clicking anywhere on the black could close the light box. But this light box is very temporary so I'm not too worried about it. Up to you @mmmavis

**Comment by cadecairos** [comment]
Yeah, we should consider this just a prototype - although I bet we'll get lots of feedback about it from our early adopters :). 

I also just realized I should try loading the page with a bogus project id in the query string. You end up with just the lightbox without content. Is there a better way of handling this?

**Comment by mmmavis** [comment]
@cadecairos what a thorough QA 😄 !

> Looks good, works as intended. If I could make one comment, it would be about how I took ten seconds to find the close button for the lightbox (top right of screen). I personally expected it to be near the card.

> I might also suggest binding a close action to the escape key or even a click event outside the card view could close it.

100% agree on this! I thought about fixing it but got lazy... since the modals are gonna be gone next week when I tackle the implementation tickets - detail view will be shown on regular page view instead (not modal).

> I also just realized I should try loading the page with a bogus project id in the query string. You end up with just the lightbox without content. Is there a better way of handling this?

@kristinashu should we display a "project not found" sorta messages?

**Comment by kristinashu** [comment]
Yeah, for now lets just have a message and a link to the landing screen.

**Comment by mmmavis** [comment]
Cooool, I filed a ticket to handle the "not found" view: https://github.com/mozilla/network-pulse/issues/73

**Comment by cadecairos** [comment]
awesome, R+

**Comment by mmmavis** [comment]
🎉 💐 🍸 🎊

**Pull request merged**